src/main/java/edu/miu/cs545/project/repository/StudentDirectoryRepo.java

```diff
@@ -4,6 +4,27 @@
 import org.springframework.data.repository.ListCrudRepository;
 import org.springframework.stereotype.Repository;
 
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
 @Repository
 public interface StudentDirectoryRepo extends ListCrudRepository<StudentDirectory, Long> {
+    Optional<List<StudentDirectory>> findByMajor(String major);
+    Optional<List<StudentDirectory>> findByAcademicYear(int date);
+
+    Optional<List<StudentDirectory>> findByUser_FirstNameOrUser_LastNameOrUser_UsernameOrUserEmailOrUser_PhoneOrUser_AddressOrUser_CityOrUser_StateOrUser_ZipOrUser_CountryOrUser_Department(String firstName,
```

Comment:
@BtsuanEyakem can we use Critieria Builder to do the search ?
https://www.baeldung.com/spring-data-criteria-queries